### PR TITLE
MHV-48208: Chem/hem data gap fixes

### PR DESCRIPTION
**Pull request by mattwrightva**
## Summary
The following fields have been updated for chem/hem (labs and tests):
#### Chem/hem
- Type of test
- Site or sample tested
- Ordered by
- Collecting location
- Lab comments
#### Chem/hem Results
- Performing lab location

## Related issue(s)
https://jira.devops.va.gov/browse/MHV-48208

> ### Implement front end data gap fixes for Chem/Hem
> Description
> Implement front end data gap fixes for Chem/Hem
> 
> Ensure that the frontend is properly mapped to the correct FHIR fields for this domain. Implement in the following locations:
> 
> web
> print
> pdf
> txt
> User Story: As a Medical Records User, I want to see the correct data in the associated fields.
> 
> GIVEN
> WHEN
> THEN
> 
> Feature Flag Y/N ? N
> Manual Testing Y/N ? Y-Rakesh
> Automated Testing Y/N ? N
> Accessibility Testing Y/N ? N
> UCD Validation Y/N N

## Testing done
Unit tests and e2e (cypress) tests were written and fixed based on new field mappings. 

## Screenshots
Screenshot 2024-05-21 at 8 30 53 PM
Screenshot 2024-05-21 at 8 31 04 PM

## What areas of the site does it impact?
MHV Medical records - labs and tests - chemistry and hematology

## Acceptance criteria
AC1 Make sure all the fields listed in the comments are mapped appropriately
AC2 Don't access the contained resources directly – do a lookup via performer/reaction respectively (see allergy for reference).

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/mhv-medical-records/reducers/labsAndTests.js [hunk 1]
```diff
@@ -38,11 +39,20 @@ const initialState = {
...
+  }
+  return null;
+};
+
 /**
  * @param {Object} record - A FHIR chem/hem Observation object
  * @returns the appropriate frontend object for display
  */
-const convertChemHemObservation = results => {
+const convertChemHemObservation = (results, record) => {
```
Comment (mmoyer-va) [inline src/applications/mhv-medical-records/reducers/labsAndTests.js:62]: The results are contained within the record. I would just pass record and pull results out here if this is the only place we need them.

File: src/applications/mhv-medical-records/reducers/labsAndTests.js [hunk 2]
```diff
@@ -76,21 +117,24 @@ const convertChemHemRecord = record => {
   const results = record.contained.filter(
     recordItem => recordItem.resourceType === 'Observation',
   );
+  const serviceRequest = extractContainedResource(
+    record,
+    record.basedOn[0]?.reference,
```
Comment (mmoyer-va) [inline src/applications/mhv-medical-records/reducers/labsAndTests.js:121]: We should verify that this array has elements before referencing one.

File: src/applications/mhv-medical-records/reducers/labsAndTests.js [hunk 3]
```diff
@@ -61,13 +71,44 @@ const convertChemHemObservation = results => {
...
       status: result.status || EMPTY_FIELD,
-      labLocation: result.labLocation || EMPTY_FIELD,
+      labLocation: getLabLocation(result.performer, record) || EMPTY_FIELD,
       labComments:
         (isArrayAndHasItems(result.note) && result.note[0].text) || EMPTY_FIELD,
     };
   });
 };
 
+const distilChemHemNotes = extension => {
```
Comment (mmoyer-va) [inline src/applications/mhv-medical-records/reducers/labsAndTests.js]: Distill has two l's 🙂

File: src/applications/mhv-medical-records/reducers/labsAndTests.js [hunk 4]
```diff
@@ -38,11 +39,27 @@ const initialState = {
...
+const getLabLocation = (performer, record) => {
+  if (isArrayAndHasItems(performer)) {
+    const locationRef = performer[0]?.reference;
+    const labLocation = extractContainedResource(record, locationRef);
+    return labLocation?.name;
+  }
+  return null;
+};
+
+const distilChemHemNotes = (notes, valueProp) => {
```
Comment (mmoyer-va) [inline src/applications/mhv-medical-records/reducers/labsAndTests.js:51]: Distill has two l's 🙂

File: src/applications/mhv-medical-records/tests/fixtures/chemHem2.json [hunk 5]
```diff
@@ -0,0 +1,317 @@
...
+          "value" : "14934-VA552"
+        }
+      ],
+      "name" : [
+        {
+          "family" : "HALL",
+          "given" : [
+            "LISA",
+            "A"
+          ]
```
Comment (mmoyer-va) [inline src/applications/mhv-medical-records/tests/fixtures/chemHem2.json:66]: Can you change this to "Jane A Doe" or something similarly generic? Same if there are any other names in here I missed.

**Review by mmoyer-va — CHANGES_REQUESTED** [review]

File: src/applications/mhv-medical-records/reducers/labsAndTests.js (hunk 4 above)
Comment (mmoyer-va) [reply src/applications/mhv-medical-records/reducers/labsAndTests.js:51]: Apparently it can also have one l, but it kind of looks weird to me.

Comment (mattwrightva) [reply src/applications/mhv-medical-records/reducers/labsAndTests.js:51]: Oh!

File: src/applications/mhv-medical-records/reducers/labsAndTests.js (hunk 2 above)
Comment (mattwrightva) [reply src/applications/mhv-medical-records/reducers/labsAndTests.js:121]: Definitely!

File: src/applications/mhv-medical-records/tests/fixtures/chemHem2.json (hunk 5 above)
Comment (mattwrightva) [reply src/applications/mhv-medical-records/tests/fixtures/chemHem2.json:66]: Yes!

**Review by mmoyer-va — APPROVED** [review]
Looks good!

File: src/applications/mhv-medical-records/reducers/labsAndTests.js (hunk 1 above)
Comment (mattwrightva) [reply src/applications/mhv-medical-records/reducers/labsAndTests.js]: Good point!

**Review by ataker — APPROVED** [review]

**Pull request merged**